jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -450,13 +450,14 @@ public ActiveSpan startActive() {
     private final Sampler sampler;
     private final Reporter reporter;
     private final PropagationRegistry registry = new PropagationRegistry();
-    private Metrics metrics;
+    private Metrics metrics = new Metrics(new StatsFactoryImpl(new NullStatsReporter()));
     private String serviceName;
     private Clock clock = new SystemClock();
     private Map<String, Object> tags = new HashMap<String, Object>();
     private boolean zipkinSharedRpcSpan;
     private ActiveSpanSource activeSpanSource = new ThreadLocalActiveSpanSource();
-    private BaggageRestrictionManager baggageRestrictionManager = new DefaultBaggageRestrictionManager();
+    private BaggageRestrictionManager baggageRestrictionManager =
+        new DefaultBaggageRestrictionManager(this.metrics);
```

Comment:
This introduces a bug; the instance of metrics used by `DefaultBaggageRestrictionManager` would always be the one set in L453.